internal/ls/symbols.go

```diff
@@ -240,17 +240,17 @@ type DeclarationInfo struct {
 
 func ProvideWorkspaceSymbols(ctx context.Context, programs []*compiler.Program, converters *lsconv.Converters, query string) (lsproto.WorkspaceSymbolResponse, error) {
 	// Obtain set of non-declaration source files from all active programs.
-	var sourceFiles collections.Set[*ast.SourceFile]
+	sourceFiles := map[tspath.Path]*ast.SourceFile{}
 	for _, program := range programs {
 		for _, sourceFile := range program.SourceFiles() {
 			if !sourceFile.IsDeclarationFile {
-				sourceFiles.Add(sourceFile)
+				sourceFiles[sourceFile.Path()] = sourceFile
 			}
 		}
 	}
 	// Create DeclarationInfos for all declarations in the source files.
 	var infos []DeclarationInfo
-	for sourceFile := range sourceFiles.Keys() {
+	for _, sourceFile := range sourceFiles {
```

Comment:
Does this end up causing the infos to be randomly ordered? Or do we sort this somewhere else?

Reply:
Ah, duh!